### DISCUSSION TITLE
Welcome to arduino-esp32 Discussions!

### DISCUSSION THREAD

**Discussion by VojtechBartoska** [espressif/arduino-esp32#6195] [Announcements]
## 👋 Welcome!
We’re using Discussions as a place to connect with other members of our community. 

As a Main Communication channel is still used [Gitter](https://gitter.im/espressif/arduino-esp32). ⭐ 

The purpose for this Discussion page is for now collect and track testing feedback for ESP32-S3 Support. 💥 

We will be glad if you help us with your contribution . ✌️

**Comment by thakursaajid** [comment]
Hello Sir,
I can not find esp32 library in the manager Libraries search
I follow with this URL " https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html " but did not find mention esp32 library
Arduino 1.8.15

**Reply by opticaldrive** [reply]
esp32 library?

It should be the boards manager....

**Comment by PratikBhagat22** [comment]
hello,
I am using ubuntu as arduino ide 2.0 run faster in ubuntu
i was using arduino 2.0.2 and esp32 packages 2.0.5 .  all was working great but some how for today i am not able to compile the esp32 codes.
getting error[  esp32 cp: target 'serial.ino.bootloader.bin' is not a directory](esptool: error: unrecognized arguments: serial.ino.bootloader.bin /home/pratik_successful/.arduino15/packages/esp32/hardware/esp32/2.0.5/tools/sdk/esp32/bin/bootloader_qio_80m.elf)


the above error not present if i use esp32 package below 2.0.0

kindly help

**Reply by me-no-dev** [reply]
Looks like something was cached. Did you restart the IDE after updating? We no longer supply the bootloader bins but rather convert the elfs we now provide.

**Comment by HeadHodge** [comment]
hi,

im using your usb-cdc client example with my esp32-s3-otg-usb connected to my win10 laptop.

it works fine with a windows terminal program connected to it. but hangs when using a win program that requires hardware handshaking.

is there a way to enable hardware handshaking on my esp32-s3? if so how?

thanks

**Comment by nileshvyas** [comment]
Hi Everyone, 

I'm currently working on a project where I want to read 4-channel 32-bit PCM audio data through the I2S interface on an ESP32-S3 MCU in Arduino IDE environment. I have an ADAU7118 evaluation board (https://www.analog.com/en/design-center/evaluation-hardware-and-software/evaluation-boards-kits/eval-adau7118.html) that converts data from 4 PDM mics into PCM and feeds it through I2S.
From what I've gathered, the default I2S library in arduino-esp32 predominantly supports 2-channel (stereo) mode. However, my application necessitates reading data from 4 mics, and I'm finding it challenging to configure the I2S to handle this 4-channel data.
Code: 
// Include I2S driver
#include <driver/i2s.h>
#define I2S_WS 6
#define I2S_SD 4
#define I2S_SCK 5
#define BITS_PER_SAMPLE 32
// Use I2S Processor 0
#define I2S_PORT I2S_NUM_0

// Define input buffer length
#define bufferLen 128
int32_t sBuffer[bufferLen];

void i2s_install() {
  // Set up I2S Processor configuration
  const i2s_config_t i2s_config = {
    .mode = i2s_mode_t(I2S_MODE_MASTER | I2S_MODE_RX), 
    .sample_rate = 8000,
    .bits_per_sample = i2s_bits_per_sample_t(32),
    .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT, 
    .communication_format = i2s_comm_format_t(I2S_COMM_FORMAT_STAND_I2S),
    .intr_alloc_flags = 1, //0
    .dma_buf_count = 8,
    .dma_buf_len = bufferLen,
    .use_apll = false
  };
  i2s_driver_install(I2S_PORT, &i2s_config, 0, NULL);
}

void i2s_setpin() {
  // Set I2S pin configuration
  const i2s_pin_config_t pin_config = {
    .bck_io_num = I2S_SCK,
    .ws_io_num = I2S_WS,
    .data_out_num = -1,
    .data_in_num = I2S_SD
  };
  i2s_set_pin(I2S_PORT, &pin_config);
}

void setup() {
  // Set up Serial Monitor
  Serial.begin(2000000);
  Serial.println("Sample demo for read audio data\n");
  // Set up I2S
  i2s_install();
  i2s_setpin();
  i2s_start(I2S_PORT);
  delay(500);
}

void loop() {
  int32_t buffer[2];
  size_t bytesIn = 0;
  esp_err_t result = i2s_read(I2S_PORT, (char*)buffer, BITS_PER_SAMPLE/8 * 2, &bytesIn, portMAX_DELAY);
  if(result == ESP_OK)
  {
    Serial.printf("Left: %d, Right: %d\n", buffer[0], buffer[1]) ;
  }
}

Has anyone encountered a similar challenge? I'm curious if:
There's a way to modify the current I2S library or directly invoke ESP-IDF functionalities from Arduino to cater to 4-channel audio?
Anyone knows of third-party libraries or tools in the Arduino ecosystem that support multi-channel I2S on the ESP32?
Any guidance, references, or pointers would be immensely appreciated. Thanks in advance!

**Comment by SHOGO-IGUCHI** [comment]
We aim to simultaneously sample (32K) the accelerometer and microphone values using UDP.
Will there be any problems if I increase the buffer size (1460) of WiFiUdp.cpp?

**Comment by richp-tech** [comment]
![image](https://github.com/espressif/arduino-esp32/assets/59874838/beac3480-28cb-4336-8f22-9c66834ca069)

Driver loaded; shows up in USB Tree; MacBook Air; won't mount the LilyGo T Display. Any Help gratefully received.

**Comment by ArmandAanekre** [comment]
I am using a ESP32-S3 in my project. It's the simplest variant with only 4MB flash. MPN: ESP32-S3-WROOM-1U-N4
When programming with PlatformIO it works OK, but when using Arduino I have problems with version 2.0.15 and 2.0.16.
Rolling back to 2.0.14 solves the problem.

I am programming using native USB with:  USB CDC On Boot: "Enabled"
When using 2.0.15/16 I have to set the chip in "download" mode with GPIO0 to be able to reprogram.
With 2.0.14 I can reprogram the chip without issues. I can even leave the Serial Monitor open without issues.

Using 2.0.15/16 there seems to be a reset missing. If I leave the Serial Monitor open, it cannot find the USB port and gives the message:

A fatal error occurred: Could not open COM17, the port doesn't exist
Failed uploading: uploading error: exit status 2

If I first close the Serial Monitor there is no response from the chip and I get the following message:
A fatal error occurred: No serial data received.
Failed uploading: uploading error: exit status 2

This has been verified with the exact same setup several times swithing between the versions.

A clue might be this yellow WARNING I get in PlatformIO when enabling verbose upload:
TimeoutError: Could not automatically find serial port for the `ESP32-S3-2DEA-1-N4` board based on the declared HWIDs=['0X303A:1001']

After the warning it autodetects COM17 and continues the upload.

**Reply by me-no-dev** [reply]
yup. we know and have found the case of the issue. Will be soon fixed in 2.0.x

**Reply by ArmandAanekre** [reply]
Perfect:)
